frontend/ui/menusorter.lua

```diff
@@ -86,7 +86,10 @@ function MenuSorter:sort(item_table, order)
                 if v then
                     if v.id == separator_id then
                         new_index = new_index - 1
-                        menu_table[order_id][new_index].separator = true
+                        -- ignore separator if the menu starts with it
+                        if new_index > 0 then
```

Comment:
Yes, I think it'd be more complex to also do something about the -/+ 1 logic then to just stick in that exception. I guess I'll leave it as is.